File: test/list.test.js
const { describe, it } = require("node:test");
const assert = require("node:assert/strict");
const gitModule = require("../lib/git");
const configModule = require("../lib/config");

describe("list", () => {
  it("calls gitInBare with 'worktree list' and logs output", (t) => {
    const worktreeOutput =
      "/projects/myrepo/main  abc1234 [main]\n/projects/myrepo/feat  def5678 [feat]";

    t.mock.method(configModule, "loadConfig", () => ({
      root: "/projects/myrepo",
      config: {},
    }));
    t.mock.method(gitModule, "gitInBare", () => worktreeOutput);
    t.mock.method(console, "log", () => {});

    delete require.cache[require.resolve("../lib/commands/list")];
    const list = require("../lib/commands/list");

    list();

    assert.equal(
      gitModule.gitInBare.mock.calls[0].arguments[0],
      "worktree list"
    );
    assert.equal(
      gitModule.gitInBare.mock.calls[0].arguments[1],
      "/projects/myrepo"
    );
    assert.equal(console.log.mock.calls[0].arguments[0], worktreeOutput);
  });
});

File: internal/cmd/list_test.go
package cmd

import (
	"strings"
	"testing"

	"github.com/gradyholmes/wt/internal/config"
)

func TestListCmd(t *testing.T) {
	t.Run("calls gitInBare with worktree list and prints output", func(t *testing.T) {
		worktreeOutput := "/projects/myrepo/main  abc1234 [main]\n/projects/myrepo/feat  def5678 [feat]"

		mg := &MockGit{
			GitInBareFunc: func(args, root string) (string, error) {
				if args != "worktree list" {
					t.Errorf("unexpected git args: %s", args)
				}
				if root != "/projects/myrepo" {
					t.Errorf("unexpected root: %s", root)
				}
				return worktreeOutput, nil
			},
		}
		mc := &MockConfig{Root: "/projects/myrepo", Cfg: &config.Config{}}
		d, stdout, _ := testDeps(mg, mc, nil)

		cmd := ListCmd(d)
		cmd.SetArgs([]string{})
		if err := cmd.Execute(); err != nil {
			t.Fatal(err)
		}

		got := strings.TrimSpace(stdout.String())
		if got != worktreeOutput {
			t.Errorf("got %q, want %q", got, worktreeOutput)
		}
	})
}
